app/models/miq_widget_set.rb

```diff
@@ -14,6 +33,33 @@ def self.with_users
     where.not(:userid => nil)
   end
 
+  def update_members
+    replace_children(Array(set_data_widgets))
+    current_user = User.current_user
+    members.each { |w| w.create_initial_content_for_user(current_user.userid) } if current_user # Generate content if not there
```

Comment:
For an update where the widget is already all set for the user, will `create_initial_content_for_user()` modify the member and cause an unnecessary save? or is it smart enough to not change it.

I never trust serialized values changed? values in rails 5.x (or maybe this wasn't fixed until 6.1)

Reply:
> For an update where the widget is already all set for the user, will `create_initial_content_for_user()` modify the member and cause an unnecessary save? or is it smart enough to not change it.
> 
> I never trust serialized values changed? values in rails 5.x (or maybe this wasn't fixed until 6.1)

`create_initial_content_for_user` contains report generation and it doesn't matter if we are re-generating it with same members - content of report generation could be different independently on members.

and `replace_children(Array(set_data_widgets))` is not needed execute when 
`members.map(&:id) == set_data_widgets.ids` so I fixed it.

> I never trust serialized values changed? values in rails 5.x (or maybe this wasn't fixed until 6.1)

I am not sure how do you mean this, I don't see usage serialize attributes here.  

thanks